src/components/dashboard/WhoContent.native.js

```diff
@@ -0,0 +1,196 @@
+  const [contacts, setContacts] = React.useState([])
+  const [initialList, setInitalList] = React.useState(contacts)
+  const [recentFeedItems, setRecentFeedItems] = React.useState([])
+  const [recentlyUsedList, setRecentlyUsedList] = React.useState([])
+
+  const getUserFeed = async () => {
+    const userFeed = await userStorage.getFeedPage(5, true)
+    const recent = userFeed.filter(({ type }) => type === 'send' || type === 'receive')
+    setRecentFeedItems(recent)
+  }
+
+  const showPermissionsAndroid = () => {
+    PermissionsAndroid.request(PermissionsAndroid.PERMISSIONS.READ_CONTACTS, {
+      title: 'Contacts',
+      message: 'We need access to view your contacts, so you can easily send G$ to them.',
+      buttonPositive: 'Approve',
+    })
+  }
+
+  useEffect(() => {
+    getUserFeed()
+  }, [])
+
+  useEffect(() => {
+    if (Contacts) {
+      if (isAndroid) {
+        showPermissionsAndroid()
+      }
+      Contacts().then((contacts, err) => {
+        if (err === 'denied') {
+          console.warn('permissions denied')
+        } else {
+          const sortContacts = orderBy(contacts, ['givenName'])
+          setContacts(sortContacts)
+          setInitalList(sortContacts)
+        }
+      })
+    }
+  }, [])
+
+  const filterContacts = useCallback(
+    memoize(searchQuery => {
+      const query = searchQuery.toLocaleLowerCase()
+
+      return initialList.filter(({ givenName, familyName, phoneNumbers }) => {
+        const [phoneNumber] = map(phoneNumbers, 'number')
+
+        return [givenName, familyName, phoneNumber].some(field => (field || '').toLocaleLowerCase().includes(query))
+      })
+    }),
+    [initialList]
+  )
+
+  const handleSearch = useCallback(
+    query => {
+      if (state) {
+        setValue(query)
+      }
+
+      setContacts(filterContacts(query))
+    },
+    [setContacts, setValue, filterContacts]
+  )
+
+  useEffect(() => {
+    if (contacts.length === 0) {
+      showNext(true)
+    } else {
+      showNext(false)
+    }
+  }, [contacts])
+
+  useEffect(() => {
+    findRecentlyUsed()
+  }, [contacts, recentFeedItems])
+
+  const findRecentlyUsed = () => {
+    let matches = []
+    if (recentFeedItems.length > 0 && contacts.length > 0) {
+      recentFeedItems.forEach(feedItem => {
+        matches.push(
+          contacts.filter(({ phoneNumbers }) => {
+            const [contactNumber] = map(phoneNumbers, 'number')
+            const { phoneNumber } = feedItem.data
+            return contactNumber && phoneNumber === contactNumber
+          })
+        )
+      })
+    }
+
+    const recentlyContacts = uniq(matches.flat())
+
+    setRecentlyUsedList(recentlyContacts)
+  }
+
+  const renderItem = useCallback(
+    horizontalMode => ({ item, index }) => {
+      return <FeedContactItem contact={item} selectContact={setContact} horizontalMode={horizontalMode} index={index} />
+    },
+    [setContact]
+  )
+
+  const ItemSeparator = () => <Separator color={styles.separatorColor} />
```

Comment:
this doesn't changes anything - it's still new function each new render
to gain something you should wrap it to use call back or move out of the component as i recommended

see https://github.com/GoodDollar/GoodDAPP/pull/1453#discussion_r413144516